backend/src/transaction_event/protocol_level_tokens.rs

```diff
@@ -641,23 +653,58 @@ impl PreparedTokenUpdate {
         .execute(tx.as_mut())
         .await?;
 
+        self.update_metrics_plt_cumulative_event_count(tx, transaction_index).await?;
+
         match self.event_type {
             TokenUpdateEventType::Mint => {
                 if let Some(ref target) = self.target {
+                    let account_exists = self.check_if_account_exists(tx, target).await?;
+
                     self.update_total_minted(tx).await?;
                     self.update_account_balance(tx, target, &self.plt_amount_change).await?;
+                    if !account_exists {
+                        self.update_metrics_plt_unique_account_count(tx, transaction_index).await?;
+                    }
                 }
             }
             TokenUpdateEventType::Burn => {
                 if let Some(ref target) = self.target {
+                    let account_exists = self.check_if_account_exists(tx, target).await?;
+
                     self.update_total_burned(tx).await?;
                     self.update_account_balance(tx, target, &(-&self.plt_amount_change)).await?;
+                    if !account_exists {
+                        self.update_metrics_plt_unique_account_count(tx, transaction_index).await?;
+                    }
                 }
             }
             TokenUpdateEventType::Transfer => {
                 if let (Some(ref from), Some(ref to)) = (&self.from, &self.to) {
+                    let from_account_exists = self.check_if_account_exists(tx, from).await?;
+                    let to_account_exists = self.check_if_account_exists(tx, to).await?;
+
                     self.update_account_balance(tx, from, &(-&self.plt_amount_change)).await?;
                     self.update_account_balance(tx, to, &self.plt_amount_change).await?;
+                    if !from_account_exists {
+                        self.update_metrics_plt_unique_account_count(tx, transaction_index).await?;
+                    }
+                    if !to_account_exists {
+                        self.update_metrics_plt_unique_account_count(tx, transaction_index).await?;
+                    }
```

Comment:
Sending queries to the database results in a lot of overhead, so we want to do as much as possible in each query.
I think the above can results in 8 queries, but I imagine this could have been 2 instead.